### PR TITLE
Enabling API key auth in Copilot FE

### PR DESCRIPTION
**Pull request by teresaqhoang**
### Motivation and Context

This PRs adds a new variable to the .env file called SK_API_KEY.

If it is present and not blank, pass its value with all requests to the Semantic Kernel server as the value of a header named `x-sk-api-key`

image

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [X] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Review by hathind-ms — APPROVED** [review]

**Comment by teresaqhoang** [comment]
Waiting on README details from @glahaye

File: samples/apps/copilot-chat-app/webapp/src/libs/services/BaseService.ts [hunk 1]
```diff
@@ -28,9 +28,14 @@ export class BaseService {
             'Content-Type': 'application/json',
         });
 
-        // For each enabled plugin, pass its auth information as a customer header
-        // to the backend so the server can authenticate to the plugin
+        // API key auth for private hosted instances
+        if (process.env.REACT_APP_SK_API_KEY) {
+            headers.append(`x-sk-copilot-api-key`, process.env.REACT_APP_SK_API_KEY as string);
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/webapp/src/libs/services/BaseService.ts:33]:
```suggestion
            headers.append(`x-sk-api-key`, process.env.REACT_APP_SK_API_KEY as string);
```

Comment (teresaqhoang) [reply samples/apps/copilot-chat-app/webapp/src/libs/services/BaseService.ts]: Was literally in process of pushing the change haha

**Review by glahaye — CHANGES_REQUESTED** [review]
Can you also change the name of the header we look for in
https://github.com/microsoft/semantic-kernel/blob/c89243acc0706c692bee8121b305fc000125f8d8/samples/apps/copilot-chat-app/webapi/Auth/ApiKeyAuthenticationHandler.cs#L20

**Review by glahaye — APPROVED** [review]

**Review by glahaye — CHANGES_REQUESTED** [review]
Actually, there is one more reference to x-api-key (as opposed to x-sk-api-key) in the README.
Could you change that too?

https://github.com/microsoft/semantic-kernel/blob/main/samples/apps/copilot-chat-app/webapi/DeploymentTemplates/README.md

**Review by glahaye — APPROVED** [review]

**Review by hathind-ms — APPROVED** [review]

**Pull request merged**